tests/rptest/scale_tests/many_topics_test.py

```diff
@@ -203,6 +222,65 @@ def acked(err, msg):
 
         return next_topic_batch, new_topic_names, errors
 
+    def _consume_messages_from_topics(self,
+                                      kclient,
+                                      message_count,
+                                      topic_names,
+                                      timeout_sec=3000):
+        """Creates a single consumer that will try to fetch all topics in 
+        `topic_names` in a single request
+
+        Args:
+            kclient (_type_): python_librdkafka client
+            message_count (_type_): number of messages to consume
+            topic_names (_type_): list of topic names to consume from
+            timeout_sec (int, optional): Timeout. Defaults to 300.
+        """
+        consumer_extra_config = {
+            "auto.offset.reset": "earliest",
+            "group.id": "topic_swarm_group_1",
+            "message.max.bytes": 1000000000,
+            "receive.message.max.bytes": 2147483647,
+        }
+
+        consumer = kclient.get_consumer(consumer_extra_config)
+        start_time = time.time()
+        total_messages_consumed = 0
+
+        try:
+            consumer.subscribe(topic_names)
+            while True:
+                elapsed = time.time() - start_time
+                if elapsed >= timeout_sec:
+                    break
+
+                res = consumer.poll(timeout=1.0)
+                # On error, check for the EOF
+                if res.error():
+                    if res.error().code() == KafkaError._PARTITION_EOF:
+                        # End of partition event
+                        self.logger.info(f"Consumer of '{res.topic()}' "
+                                         f"[{res.partition()}] reached "
+                                         f"end at offset {res.offset()}")
+                        break
+                    # If not EOF, raise it
+                    elif res.error():
```

Comment:
`res.error()` has already been checked by the outer condition, so this should just be `else:`? Also explains why the else condition is missing.

Reply:
Correct, the condition should've just been; 
```python
if res.error() and res.error().code() != KafkaError._PARTITION_EOF:
    raise KafkaException(res.error())
```